examples/linear_elasticity/test_elasticity.py

```diff
@@ -213,6 +240,35 @@ def right_boundary(x):
         )
         > 1e-2
     )
+    law_3d = LinearElasticityModel(
```

Comment:
Instead of adding a variabe to the interface of all models, wouldn't it be easier to create a dummy class MaterialModelUniaxialStrainFrom3D or MaterialModelPlaneStrainFrom3D that takes as input a 3D model and can be used in 1D. Then we do not have to pollute the interface.

Reply:
As I understand it, only the solver gets an additional optional parameter.

Reply:
Maybe, but then I would need to transform all the input-vectors

```python
class UniaxialStrainFrom3D:
    def __init__(self, model):
        assert model.constraint == Constraint.FULL
        self.3d_model = model

    def evaluate(
        self,
        time: float,
        del_t: float,
        grad_del_u: np.ndarray,
        mandel_stress: np.ndarray,
        tangent: np.ndarray,
        history: np.ndarray | dict[str, np.ndarray] | None,
    ) -> None:
        3d_grad_del_u = as_3d_tensor(grad_del_u)
        3d_stress = as_3d_stress(mandel_stress)
        3d_tangent = as_3d_tangent(tangent)
        self.3d_model.evaluate(time, del_t, 3d_grad_del_u,...)
        mandel_stress[:] = as_1D_stress(3d_stress)
        tangent[:] = as_1D_tangent(3d_tangent)

```

This would leave the solver clean, but it would generate extra data.

Reply:
But thinking about it, this might be the nicer approach, since I hope that we will write 1D models as 1D models where possible and hopefully not use these wrapper-classes a lot. And in that case it's micer to have the cleaner interface in the solver